### ISSUE TITLE
Songs with the same Artist  MBID but different artist name show the same name

### ISSUE THREAD

**Issue by cquike** [ampache/ampache#3542]
## Description

If I have two songs which have the same artist mbid but the name of the artist is credited differently in both songs. Musicbrainz has support for this use case. However, when importing the two songs in ampache the name of the artist is always as the credited name in the first song that was imported, leading to the wrong credit in the second song.

### To reproduce

1. Add two songs with the same mbid for the artist but with different artist name
2. Check the artist names for the two songs in Ampache
3. 
### Expected behavior

 I think the name shown for each song (or album) should be what is credited for. However clicking on that artist would still show all the songs and albums belonging to the same artist (identified by the MBID).

Again Musicbrainz is probably a good model to follow. For instance, for this artist:
https://musicbrainz.org/artist/fe7245e7-d734-4ca1-8e26-691883f58201

In the list of songs, sometimes the artist is credited as "Miles Davis Quintet" , sometimes "The Miles Davis Quintet" and sometimes "The New Miles Davis Quintet".  So I guess there should be provision in the song/album tables to store the name of the artist as credited 

The thing is probably a bit more complicated gi

## Environment

* Ampache version:
* Web server + version:
* Server operating system:
* Client operating system:

### Client type

* Web
* Ampache
* SubSonic
* DAAP
* UPnP
* WebDAV

**Comment by Nizhile** [comment]
The problem might be to rely on artist mbid to create artist. I get a similar issue where a compilation was using some native names for classical artists. Consequence : I cannot find out a French release of Tchaikovsky n°6 conducted by Seiji Ozawa except by its title 🤪 as it is from Пётр Ильич Чайковский, 小澤征爾

**Comment by cquike** [comment]
So this probably require changes in the database. Right now there is a single artist table and the songs get relationships with that table. There is currently no field to specify the credited artist name, since it will always use the name in the artist table, which is common to all the songs of the same artist and provided from MusicBrainz.
Maybe a field can be added to the artist_map table as the credited artist. If different from NULL, then it will show that name in a given song or album. If NULL, then it uses the "general" name of the artist as it appears in the artist table.
Searches, like in your case in which you want to search for the name as it appears in a given release, would probably be more expensive, since there would be a need to search for an artist both in the artist table and in the artist_map table, effectively doubling the search time, but I guess still doable and within reason.

**Comment by Nizhile** [comment]
One possible approach would be to simply duplicate the artist even if mbid already exists in database. Search would have the same cost. On ampache artist page, the also know as can maybe provided.
I do not know usage (either fixing typos or whatever present in tags) but if tags contains mbid, chances are that the problem is coming directly from musicbrainz.

**Comment by cquike** [comment]
Yes, that's an option, to add duplicate entries for the same artist. However in this case, which of the entries should be used when displaying info about an artist? Also, extra logic should be added to make sure that when you are just interested in an "artist" (independently of which albums it has associated) you display the proper entry of all the possible duplications.
